src/datasources/cache.ts

```diff
@@ -0,0 +1,50 @@
+import Keyv from "keyv";
+import KeyvRedis from "@keyv/redis";
+import Redis from "ioredis";
+import { KeyvAdapter } from "@apollo/utils.keyvadapter";
+import { cacheConfig } from "../config/cacheConfig";
+import { logger, formatLogMessage } from '../logger';
+
+export class Cache {
+  private static instance: Cache;
+  public adapter: KeyvAdapter;
+
+  private constructor() {
+    // Setup the Redis Cluster
+    const cluster = new Redis.Cluster(cacheConfig.cluster);
+
+    // Having trouble figuring how how to type `Keyv` as `Keyv<string, Record<string, any>>`
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const keyV = new Keyv(new KeyvRedis(cluster)) as any;
+
+    keyV.on('connect', () => {
+      formatLogMessage(logger).error(null, `Redis connection established`);
```

Comment:
L21 and L29: For these cases, would you want to use ".info()" instead of ".error()"?

Reply:
yes. good catch